core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLTPFSTestRun.java

```diff
@@ -220,6 +239,64 @@ public void testAvroSourceConversionToAvroSink() throws Exception {
   }
 
   @Test
+  public void testOrc() throws Exception {
+    Schema recordSchema = Schema.recordOf("record",
+                                          Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+                                          Schema.Field.of("address", Schema.of(Schema.Type.STRING)),
+                                          Schema.Field.of("phone", Schema.of(Schema.Type.STRING)),
+                                          Schema.Field.of("contact", Schema.of(Schema.Type.STRING))
+    );
+
+    ETLPlugin sinkConfig = new ETLPlugin("TPFSOrc",
+                                         BatchSink.PLUGIN_TYPE,
+                                         ImmutableMap.of(
+                                           "schema", recordSchema.toString(),
+                                           "name", "outputOrc"),
+                                         null);
+
+    ETLStage sink = new ETLStage("sink", sinkConfig);
+    String inputDatasetName = "input-batchsinktest";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputDatasetName));
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "TPFSOrcSinkTest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    Schema recordSchema2 = Schema.recordOf("record",
+                                           Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+                                           Schema.Field.of("address", Schema.of(Schema.Type.STRING)),
+                                           Schema.Field.of("phone", Schema.of(Schema.Type.STRING)),
+                                           Schema.Field.of("contact", Schema.of(Schema.Type.STRING))
+    );
+
+    // write input data
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(recordSchema2)
+        .set("name", "a")
+        .set("address", "b")
+        .set("phone", "c")
+        .set("contact", "d").build()
+    );
+    MockSource.writeInput(inputManager, input);
+
+    // run the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(4, TimeUnit.MINUTES);
+
+    Connection connection = getQueryClient();
+    ResultSet results = connection.prepareStatement("select * from dataset_outputOrc").executeQuery();
+    results.next();
+    Assert.assertFalse(results == null);
```

Comment:
this can never be null because you have called next() already.  I think you want to do `Assert.assertTrue(results.next())` above.